Enforce Pricing for Team Sharing

hotzenklotz opened the pull request:
PR adds a new wrapper component blur components/features that are not part of an organization's selected plan.

Changes:
- new blurring wrapper component
- affects Team sharing modal and dataset setting's sharing tab
- moved `PricingPlanEnum` to `pricing_plan_utils` to unroll a cyclic dependency 

### URL of deployed dev instance (used for testing):
- https://moreplans.webknossos.xyz/

### Steps to test:
- Set orga plan accordingly in your DB
- Check annotation team sharing modal
- Check dataset sharing tab

### Issues:
- contributes to #6509 

------
(Please delete unneeded items, merge only when none are left open)
- [x] Updated [(unreleased) changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [ ] Updated [(unreleased) migration guide](../blob/master/MIGRATIONS.unreleased.md#unreleased) if applicable
- [ ] Updated [documentation](../blob/master/docs) if applicable
- [ ] Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes
- [ ] Adapted [wk-libs python client](https://github.com/scalableminds/webknossos-libs/tree/master/webknossos/webknossos/client) if relevant API parts change
- [ ] Needs datastore update after deployment
- [x] Ready for review

philippotto commented on frontend/javascripts/components/pricing_enforcers.tsx:
```diff
@@ -75,14 +64,42 @@ export function PricingEnforcedButton({
+    state.activeOrganization ? state.activeOrganization.pricingPlan : PricingPlanEnum.Basic,
+  );
+  const isFeatureAllowed = isPricingPlanGreaterEqualThan(currentPricingPlan, requiredPricingPlan);
+  if (isFeatureAllowed) return <>children</>;
```
I think, this would render the string "children" :thinking: 

```suggestion
  if (isFeatureAllowed) return <>{children}</>;
```
or:
```
  if (isFeatureAllowed) return children;
```

hotzenklotz replied:
Ah, good catch. Unfortunately, `return children` does not work... well at least Typescript complains... (React actually was able to work with this nonetheless). Hence, I had to wrap it in a Fragment to satisfy the type checker.

philippotto reviewed (commented):
Code looks good except for maybe one thing :+1: Didn't test yet, though.

philippotto reviewed (approved):
Also works well :+1: 

However, it took me a bit until I could fully "parse" the blurred component. Especially when the blurred area is relatively large, the lock icon is very easy to miss. For example:

![image](https://user-images.githubusercontent.com/2486553/214567265-72586481-0cdd-44ef-862d-0aded080bd88.png)

![image](https://user-images.githubusercontent.com/2486553/214567500-3a9520e5-32a3-4603-9974-f10ac069552e.png)

I'm not sure how this could be improved. Initially, I had the idea to simply always show the tooltip content next to the lock icon in the center of the blurred area (including the black tooltip background). However, I don't know whether it will look good and will work with the wide teams input that is not very tall. Maybe another color for the lock icon?

Maybe you want to experiment a bit. If it stays as is, it's probably okay, too, since the tooltip will show why things are blurred, too.

By the way, during testing, my Spotify "Discover Weekly" playlist decided to play a song called "Blurry Mess". I'm feeling watched :sweat_smile:

hotzenklotz commented on frontend/javascripts/oxalis/view/action-bar/tracing_actions_view.tsx:
```diff
@@ -123,6 +123,8 @@ export function LayoutMenu(props: LayoutMenuProps) {
     setAutoSaveLayouts,
     saveCurrentLayout,
+    // rome-ignore lint/correctness/noUnusedVariables: underscore prefix does not work with object destructuring
+    setCurrentLayout,
```
fix for accidentally introduced bug in my recent Linter PR

hotzenklotz replied:
React/Antd complaining about the `setCurrentLayout` prop being present in/leaking into a `<Menu>` component, where it was not expected/supported.

hotzenklotz commented:
@philippotto Please re-review with latest changes:
- I added helper function like we discussed, see `isFeatureAllowedByPricingPlan`
- Added checks for `Pre-Computed Mesh` creation button
- Added checks for Proof-Reading tool

When testing, please try anonymous access / publicly shared annotation as well. Please also consider how else some of these features can be activated, e.g. through keyboard shortcuts.

philippotto commented:
Testing went well :+1:

One thing I noticed is that if a feature is disabled due to the incorrect orga plan, the user won't know which plan is necessary. For example, for the proofreading tool, one will have to look up the plan comparison to see that the power plan is required. One could mention this in the tooltip. However, it would just be nice to have, I think.
